apiserver/facades/client/usermanager/usermanager_test.go

```diff
@@ -821,3 +821,88 @@ func (s *userManagerSuite) TestRemoveUserBulkSharedModels(c *gc.C) {
 	c.Assert(alice.IsDeleted(), jc.IsTrue)
 
 }
+
+func (s *userManagerSuite) TestResetPassword(c *gc.C) {
+	alex := s.Factory.MakeUser(c, &factory.UserParams{Name: "alex", NoModelUser: true})
+	c.Assert(alex.PasswordValid("password"), jc.IsTrue)
+
+	args := params.Entities{Entities: []params.Entity{{Tag: alex.Tag().String()}}}
+	results, err := s.usermanager.ResetPassword(args)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(results.Results, gc.HasLen, 1)
+
+	err = alex.Refresh()
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(results.Results[0].Tag, gc.DeepEquals, alex.Tag().String())
+	c.Assert(results.Results[0].SecretKey, gc.DeepEquals, alex.SecretKey())
+	c.Assert(alex.PasswordValid("password"), jc.IsFalse)
+}
+
+func (s *userManagerSuite) TestBlockResetPassword(c *gc.C) {
+	alex := s.Factory.MakeUser(c, &factory.UserParams{Name: "alex", NoModelUser: true})
+	args := params.Entities{Entities: []params.Entity{{Tag: alex.Tag().String()}}}
+	c.Assert(alex.PasswordValid("password"), jc.IsTrue)
+
+	s.BlockAllChanges(c, "TestBlockResetPassword")
+	_, err := s.usermanager.ResetPassword(args)
+	// Check that the call is blocked
+	s.AssertBlocked(c, err, "TestBlockResetPassword")
+
+	err = alex.Refresh()
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(alex.PasswordValid("password"), jc.IsTrue)
+}
+
+func (s *userManagerSuite) TestResetPasswordForSelf(c *gc.C) {
```

Comment:
Well, originally I wanted to give ability to controller admins to change their own password but there is no reason to stop other users from changing their own pwd too.